RecoLocalTracker/SiStripClusterizer/interface/StripClusterizerAlgorithm.h

```diff
@@ -19,57 +19,82 @@ class FedChannelConnection;
-  virtual void clusterizeDetUnit(const edmNew::DetSet<SiStripDigi> &, output_t::FastFiller &) = 0;
+  void clusterize(const    edm::DetSetVector<SiStripDigi> &, output_t &) const;
+  void clusterize(const edmNew::DetSetVector<SiStripDigi> &, output_t &) const;
+  virtual void clusterizeDetUnit(const    edm::DetSet<SiStripDigi> &, output_t::FastFiller &) const = 0;
+  virtual void clusterizeDetUnit(const edmNew::DetSet<SiStripDigi> &, output_t::FastFiller &) const = 0;
 
   //HLT stripByStrip interface
-  virtual bool stripByStripBegin(uint32_t id) = 0;
+  virtual Det stripByStripBegin(uint32_t id)const  = 0;
 
-  virtual void addFed(sistrip::FEDZSChannelUnpacker & unpacker, uint16_t ipair, std::vector<SiStripCluster>& out) {}
-  virtual void stripByStripAdd(uint16_t strip, uint8_t adc, std::vector<SiStripCluster>& out) {}
-  virtual void stripByStripEnd(std::vector<SiStripCluster>& out) {}
+  virtual void addFed(Det const & det, sistrip::FEDZSChannelUnpacker & unpacker, uint16_t ipair, std::vector<SiStripCluster>& out)  const {}
+  virtual void stripByStripAdd(State & state, uint16_t strip, uint8_t adc, std::vector<SiStripCluster>& out)  const{}
+  virtual void stripByStripEnd(State & state, std::vector<SiStripCluster>& out)  const {}
+
+  virtual void addFed(State & state, sistrip::FEDZSChannelUnpacker & unpacker, uint16_t ipair, output_t::FastFiller & out)  const {}
+  virtual void stripByStripAdd(State & state, uint16_t strip, uint8_t adc, output_t::FastFiller & out)  const {}
+  virtual void stripByStripEnd(State & state, output_t::FastFiller & out)  const {}
 
-  virtual void addFed(sistrip::FEDZSChannelUnpacker & unpacker, uint16_t ipair, output_t::FastFiller & out) {}
-  virtual void stripByStripAdd(uint16_t strip, uint8_t adc, output_t::FastFiller & out) {}
-  virtual void stripByStripEnd(output_t::FastFiller & out) {}
-  
-  virtual void cleanState(){}
 
   struct InvalidChargeException : public cms::Exception { public: InvalidChargeException(const SiStripDigi&); };
 
 
   SiStripDetCabling const * cabling() const { return theCabling;} 
   std::vector<uint32_t> const & allDetIds() const { return detIds;}
-  std::vector<const FedChannelConnection *> const & currentConnection() const { return connections[ind]; }
+
+  std::vector<const FedChannelConnection *> const & currentConnection(const Det& det) const { return connections[det.ind]; }
 
  protected:
 
   StripClusterizerAlgorithm() : qualityLabel(""), noise_cache_id(0), gain_cache_id(0), quality_cache_id(0) {}
 
-  uint32_t currentId() {return detId;}
-  bool setDetId(const uint32_t);
-  float noise(const uint16_t& strip) const { return SiStripNoises::getNoise( strip, noiseRange ); }
-  float gain(const uint16_t& strip)  const { return SiStripGain::getStripGain( strip, gainRange ); }
-  bool bad(const uint16_t& strip)    const { return qualityHandle->IsStripBad( qualityRange, strip ); }
+  // uint32_t currentId() const {return detId;}
```

Comment:
Thanks for noticing.
I will remove it next round (when integrating ThreadSafe DetSetVector)